src/ZenohUTransport.cpp

```diff
@@ -273,10 +275,13 @@ v1::UStatus ZenohUTransport::sendRequest_(const std::string& zenoh_key,
 	auto on_done = []() {};
 
 	try {
+#pragma GCC diagnostic push
+#pragma GCC diagnostic ignored "-Wpedantic"
```

Comment:
Please provide documentation of the specific warning being ignored and the reason it is safe to ignore in this context.

Reply:
In all three cases, the following error was being suppressed:
error: C++ designated initializers only available with ‘-std=c++20’ or ‘-std=gnu++20’ [-Werror=pedantic]
I'm adding this as a comment into the code.
It seems impossible to narrow the pragma to inside of the put() arguments. There may be a way to demote this from an error to a warning, but the easy fix would be to switch to C++20.

Reply:
We investigated switching to C++20, but the discussion stalled: https://github.com/eclipse-uprotocol/up-cpp/issues/109

Reply:
We could also remove pedantic.  Seems silly to be holding strict to ISO C/C++.  Was there discussion about having this warning enabled?

Reply:
This use of a GCC pragma is also non-portable.  I would recommend removing it.

Reply:
Please fix this the correct way, buy removing the .references.  Put in manual assignments maybe to a temp.